tarodb/assets_store.go

```diff
@@ -1556,6 +1562,48 @@ func (a *AssetStore) LogPendingParcel(ctx context.Context,
 	})
 }
 
+// StoreParcelDeliveryAttempt logs a parcel delivery attempt to disk.
+func (a *AssetStore) StoreParcelDeliveryAttempt(ctx context.Context,
+	anchorPoint wire.OutPoint) error {
+
+	anchorPointBytes, err := encodeOutpoint(anchorPoint)
+	if err != nil {
+		return err
+	}
+
+	var writeTxOpts AssetStoreTxOptions
+	return a.db.ExecTx(ctx, &writeTxOpts, func(q ActiveAssetsStore) error {
+		// First, we'll fetch the asset transfer using the given
+		// outpoint bytes. We can then relate the delivery attempt to
+		// the asset transfer entry.
+		assetTransfers, err := q.QueryAssetTransfers(ctx, TransferQuery{
```

Comment:
If we wish, we can condense this into a single `sqlc` call, but using a `CTE` to find the target transfer ID based on the anchor point. It would then use that to know which ID to use for insertion.

Reply:
I explored that possibility with Jordi. We came to the conclusion that it was simpler to do two calls than the join the CTE would entail. Please let me think differently and would like me to go down the CTE route.